libscript/src/script-builder.cpp

```diff
@@ -1360,187 +1532,140 @@ void MCScriptResolveLabelForBytecodeInModule(MCScriptModuleBuilderRef self, uind
-{
-    if (self == nil || !self -> valid)
-        return;
-
-    __begin_instruction(self, kMCScriptBytecodeOpAssignArray);
-    __continue_instruction(self, p_reg);
-}
-
-void MCScriptEmitContinueAssignArrayInModule(MCScriptModuleBuilderRef self, uindex_t p_reg)
-{
-    if (self == nil || !self -> valid)
-        return;
-
-    __continue_instruction(self, p_reg);
-}
-
-void MCScriptEmitEndAssignArrayInModule(MCScriptModuleBuilderRef self)
-{
-    if (self == nil || !self -> valid)
-        return;
-
-    __end_instruction(self);
-}
-
-void MCScriptEmitAssignInModule(MCScriptModuleBuilderRef self, uindex_t p_dst_reg, uindex_t p_src_reg)
-{
-    if (self == nil || !self -> valid)
-        return;
-    
-    __emit_instruction(self, kMCScriptBytecodeOpAssign, 2, p_dst_reg, p_src_reg);
-}
-
-void MCScriptEmitReturnInModule(MCScriptModuleBuilderRef self, uindex_t p_src_reg)
-{
-    if (self == nil || !self -> valid)
-        return;
-    
-    __emit_instruction(self, kMCScriptBytecodeOpReturn, 1, p_src_reg);
-}
-
-void MCScriptEmitReturnUndefinedInModule(MCScriptModuleBuilderRef self)
+static MCScriptDefinitionKind __kind_of_definition(MCScriptModuleBuilderRef self, uindex_t p_index)
 {
-    if (self == nil || !self -> valid)
-        return;
+    return self -> definition_kinds[p_index];
     
-    // If the zero argument form of Return is used, then the return value is
-    // taken to be undefined.
-    __emit_instruction(self, kMCScriptBytecodeOpReturn, 0);
-}
-
-void MCScriptBeginInvokeInModule(MCScriptModuleBuilderRef self, uindex_t p_handler_index, uindex_t p_result_reg)
-{
-    if (self == nil || !self -> valid)
-        return;
+    if (self -> module . definitions[p_index] -> kind == kMCScriptDefinitionKindExternal)
+    {
+        MCScriptExternalDefinition *t_ext_definition;
+        t_ext_definition = static_cast<MCScriptExternalDefinition *>(self -> module . definitions[p_index]);
+        return self -> module . imported_definitions[t_ext_definition -> index] . kind;
+    }
     
-    __begin_instruction(self, kMCScriptBytecodeOpInvoke);
-    __continue_instruction(self, p_handler_index);
-    __continue_instruction(self, p_result_reg);
+    return self -> module . definitions[p_index] -> kind;
 }
 
-void MCScriptBeginInvokeIndirectInModule(MCScriptModuleBuilderRef self, uindex_t p_handler_reg, uindex_t p_result_reg)
+void MCScriptEmitBytecodeInModuleA(MCScriptModuleBuilderRef self, uindex_t p_opcode, uindex_t *p_arguments, uindex_t p_argument_count)
 {
     if (self == nil || !self -> valid)
         return;
     
-    __begin_instruction(self, kMCScriptBytecodeOpInvokeIndirect);
-    __continue_instruction(self, p_handler_reg);
-    __continue_instruction(self, p_result_reg);
-}
-
-void MCScriptContinueInvokeInModule(MCScriptModuleBuilderRef self, uindex_t p_arg_reg)
-{
-    if (self == nil || !self -> valid)
+    if (p_opcode >= sizeof(s_bytecode_info) / sizeof(s_bytecode_info[0]) ||
```

Comment:
I think it might be a good idea to have an `inline` function that returns this value (i.e. number of known bytecodes), since it's used in several places.